ls1intum/Artemis
# [Enhancement] exam assessment dashboards, locks & complaint dashboards

### Checklist

### Motivation and Context


The exam mode "assessment-dashboard" does only show the exercises, unlike the course "assessment-dashboard", which also has information and a leader-board. 
In this PR we take first steps to align the two dashboards views. 
Furthermore we added pages to see the complaints and assessment locks of exams. Those are only accessible for instructors at the moment.
The leader-board is also still missing, as this will need a bigger change.


### Description

<details>
<summary>Major changes:</summary>
Former course - assessment-dashboard:
<img src="https://user-images.githubusercontent.com/33342534/109975098-8df14680-7cfa-11eb-9b39-a55c17238517.png">
After change: 
<img src="https://user-images.githubusercontent.com/33342534/109975289-c98c1080-7cfa-11eb-8e86-f1fb9d1646a8.png">


Former exam - assessment-dashboard:
<img src="https://user-images.githubusercontent.com/33342534/109975451-ecb6c000-7cfa-11eb-91f5-3ef8afbec647.png">
After change: 
<img src="https://user-images.githubusercontent.com/33342534/109975624-1ff94f00-7cfb-11eb-8a56-6cd2ee6896fa.png">

</details>

Some other changes:
1. Added exam- complaint and assessment-locks pages for instructors:
![image](https://user-images.githubusercontent.com/33342534/109980773-8b91eb00-7d00-11eb-8273-f19c32fee2b2.png)
![image](https://user-images.githubusercontent.com/33342534/109980845-a1071500-7d00-11eb-9ae9-9470428f58ce.png)

2. Added "Submissions" button in exam-mode for instructors to assessment- and exercise-assessment-dashboards.
![image](https://user-images.githubusercontent.com/33342534/109976014-96964c80-7cfb-11eb-8105-098dd2e6e4f2.png)
3. added better info messages to the exercise-assessment-dashboard
<img src="https://user-images.githubusercontent.com/33342534/109976272-e37a2300-7cfb-11eb-8324-2bca1e11e7f8.png">
4. fixed sorting issues in leader-boards

### Steps for Testing


-  Check in different courses if the assessment and exercise-assessment dashboard are showing the correct things. Test both dashboards for the course, and exams.
- Test if the complaint and locks links work for the instructor.
- Check if the "submissions" buttons work in the assessment + exercise-assessment dashboard (only for exam mode atm)
- in an exam with 2 correction-rounds, go into an exercise-assessment-dashboard and check if all the messages make sense.
- check if the leader-boards are initially sorted descending by points.
